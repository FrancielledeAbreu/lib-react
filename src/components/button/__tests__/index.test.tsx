import React from 'react'
import renderer from 'react-test-renderer'
import HoverButton from '../index'
import { shallow } from 'enzyme'

describe('Snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <HoverButton>
          <p>Teste</p>
        </HoverButton>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('Events', () => {
  it('button onClick ', () => {
    const onButtonClick = jest.fn()
    const wrapper = shallow(
      <HoverButton onClick={onButtonClick}>Teste</HoverButton>
    )
    wrapper.simulate('click')

    expect(onButtonClick).toHaveBeenCalled()
  })
})
