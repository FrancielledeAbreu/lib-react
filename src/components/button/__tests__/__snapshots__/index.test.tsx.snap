// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders correctly 1`] = `
<button
  className="sc-bdnylx cfBdBq"
>
  <p>
    Teste
  </p>
</button>
`;
